### PR TITLE
Explain how the module works in the README

### PR DESCRIPTION
**Pull request by jviotti**
Signed-off-by: Juan Cruz Viotti <jviotti@openmailbox.org>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+mountpoints:
+  - path: /
+raw: /dev/rdisk1
+protected: False
+system: True
+```
+
+Because of the simplicity of this module's design, supporting a new operating
+system simply means adding a new script to `scripts/` that gathers drive data
+and output something similar to the above example. The challenge with this is
```
Comment (lurch) [inline README.md]: output -> outputs

File: README.md [hunk 2]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+that we must ensure all the platform scripts prints consistent output.
```
Comment (lurch) [inline README.md]: prints -> print

File: README.md [hunk 3]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+
+When the user executes `drivelist.list()`, the module checks what's the
```
Comment (lurch) [inline README.md]: remove " what's"

File: README.md [hunk 4]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+operating system of the client, and executes the corresponding drive scanning
+script as a child process. It then parses the YAML output of the scripts as an
```
Comment (lurch) [inline README.md]: "output of the scripts" -> "output of the script"

File: README.md [hunk 5]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+system: True
+
+device: /dev/disk1
+description: "Macintosh HD"
+size: 249779191808
+mountpoints:
+  - path: /
+raw: /dev/rdisk1
+protected: False
+system: True
```
Comment (lurch) [inline README.md]: Is this example even necessary, given that other examples are already included further below?

File: README.md [hunk 6]
```diff
@@ -17,6 +17,53 @@ Supports:
 - GNU/Linux distributions that include [util-linux](https://github.com/karelzak/util-linux) and [udev](https://wiki.archlinux.org/index.php/udev).
 - Mac OS X.
 
+The `drivelist` core consists of a set of scripts built with technologies that
+are available by default on the target operating systems (like Bash, VBScipt,
```
Comment (lurch) [inline README.md]: VBScipt -> VBScript

File: README.md [hunk 7]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+etc). Each of these scripts attempts to get information about the available
+drives and metadata related to them using any medium the target platform
+provides, like a combination of `diskutil`, `/proc/mounts` etc. You can find
+these scripts in the `scripts/` directory.
+
+The scripts then are expected to print to `stdout` all the drive information
+they gathered in a predefined way, based on the YAML language. The type of
```
Comment (lurch) [inline README.md]: Perhaps make 'YAML' a link to yaml.org ?

File: README.md [hunk 8]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+output `drivelist` expects consists of a set of blank line separated blocks,
+each representing a drive by a set of key/value pairs. The exact keys that we
+expect are constantly changing while we keep improving this module, but you can
+see what's the current expected properties by running the platform script that
+applies to your operating system.
```
Comment (lurch) [inline README.md]: applies -> corresponds

**Comment by jviotti** [comment]
Haha, amazing feedback :)

File: README.md (hunk 6 above)
Comment (jviotti) [inline README.md]: Fixed!

File: README.md [hunk 9]
```diff
@@ -17,6 +17,53 @@ Supports:
 - GNU/Linux distributions that include [util-linux](https://github.com/karelzak/util-linux) and [udev](https://wiki.archlinux.org/index.php/udev).
 - Mac OS X.
 
+The `drivelist` core consists of a set of scripts built with technologies that
+are available by default on the target operating systems (like Bash, VBScipt,
+etc). Each of these scripts attempts to get information about the available
+drives and metadata related to them using any medium the target platform
+provides, like a combination of `diskutil`, `/proc/mounts` etc. You can find
```
Comment (jviotti) [inline README.md]: Yeah, reads much better 👍

File: README.md [hunk 10]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+these scripts in the `scripts/` directory.
+
+The scripts then are expected to print to `stdout` all the drive information
```
Comment (jviotti) [inline README.md]: Fixed!

File: README.md (hunk 7 above)
Comment (jviotti) [inline README.md]: Fixed!

Comment (jviotti) [inline README.md]: Fixed!

File: README.md [hunk 11]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+they gathered in a predefined way, based on the YAML language. The type of
+output `drivelist` expects consists of a set of blank line separated blocks,
```
Comment (jviotti) [inline README.md]: Sounds good, fixed!

File: README.md [hunk 12]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+each representing a drive by a set of key/value pairs. The exact keys that we
```
Comment (jviotti) [inline README.md]: Fixed!

File: README.md [hunk 13]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+expect are constantly changing while we keep improving this module, but you can
+see what's the current expected properties by running the platform script that
```
Comment (jviotti) [inline README.md]: Fixed!

File: README.md (hunk 8 above)
Comment (jviotti) [inline README.md]: Fixed!

File: README.md [hunk 14]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+applies to your operating system.
+
+This is how the raw output looks in my MacBook Pro at the time of this writing:
```
Comment (jviotti) [inline README.md]: Fixed!

File: README.md [hunk 15]
```diff
@@ -17,6 +17,53 @@ Supports:
...
+
+```sh
+$ ./scripts/darwin.sh
+device: /dev/disk0
+description: "APPLE SSD SM0256G"
+size: 251000193024
+mountpoints: []
```
Comment (jviotti) [inline README.md]: `/dev/disk0` is always tricky to explain. The SSD is `/dev/disk1`, while `/dev/disk0` is some kind of OS X specific logical disk that includes the recovery stuff, which is not mounted anywhere.

File: README.md (hunk 5 above)
Comment (jviotti) [inline README.md]: The difference is that this one demonstrates the plain output from the platform scripts while the other show the parsed JSON version.

File: README.md (hunk 1 above)
Comment (jviotti) [inline README.md]: Fixed!

File: README.md (hunk 2 above)
Comment (jviotti) [inline README.md]: Fixed!

File: README.md (hunk 3 above)
Comment (jviotti) [inline README.md]: Fixed!

File: README.md (hunk 4 above)
Comment (jviotti) [inline README.md]: Fixed!

File: README.md (hunk 5 above)
Comment (lurch) [inline README.md]: Ah, I hadn't noticed that. I guess I must have skim-read too quickly ;-)

File: README.md (hunk 15 above)
Comment (lurch) [inline README.md]: Lol, damn Apple ;-)

**Pull request merged**